src/Tribe/Editor/Objects/Event.php

```diff
@@ -59,6 +59,17 @@ public function data( $key = null, $default = null ) {
 				'is_new_post' => true,
 			];
 
+			/**
+			 * Grabs the tribe-start-date query param from the url 
+			 * if it exists then adds it to the global window object.
+			 */
+			if ( isset( $_GET['tribe-start-date'] ) ) {
+				$this->data = [
+					'is_new_post'      => true,
+					'tribe_start_date' => $_GET['tribe-start-date'],
+				];
+			}
```

Comment:
Getting data from a user-input data is inherently unsafe, and using it directly in our application is also unsafe. A few things:

1. There's a function called `tribe_get_request_var()` (see common for the function definition) that allows us to get the variable from `$_GET` safely by sanitizing the value, use that.
2. We should be checking if the `tribe-start-date` we get is of proper format. If not, we shouldn't output it onto the data.
3. No need to redefine `$this->data` here, just define a new key and value.

Reply:
I'm now using the `tribe_get_request_var()` function to get the variable and validating it before adding it to the data.